### ISSUE TITLE
oidc: staging environment against google shows `OAuth 2 MUST utilize https`

### ISSUE THREAD

**Issue by jgehrcke** [conbench/conbench#480]
Seen yesterday when @austin3dickey and I deployed from (at that time) head of main to a staging env. Upon initiating the flow:
```
{"code": 400, "description": {"_errors": ["OpenID Connect single sign-on flow failed: (insecure_transport) OAuth 2 MUST utilize https.."]}, "name": "Bad Request"}
```

**Comment by jgehrcke** [comment]
Debugging, with the help of logs, code, etc.

What fails is the conclusion of the flow, i.e. the error response shown above is to a GET request to `/api/google/callback?code=4%2F0Afg....`.

**Comment by jgehrcke** [comment]
Corresponding access log line showing the 400 response, redacted:
```
172.31.9.214 - - [01/Dec/2022:13:30:34 +0000] "GET /api/google/callback?code=4%2F0AfgeXvu[...]ompt=none HTTP/1.1" 400 162 "https[...]/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36"
```

**Comment by jgehrcke** [comment]
While CI is running on the patch, my attempt to explain the problem:

This oauthlib method call raised an exception: https://github.com/conbench/conbench/blob/175439dfbcfc18c2c51bb4e1b43635c65207e02e/conbench/api/_google.py#L153

As of https://github.com/conbench/conbench/issues/464 even in 'production' when actual user agents reach the Conbench HTTP API via TLS, i.e. with the URL scheme `https://`, the WSGI environment might tell the app that the access happened via `http://`, so that `f.request.url` starts with the 'wrong' scheme.

Security-wise, we do not need oauthlib to do _that_ kind of validation. One way to disable is: always replace `http://` with `https://`. That is what we did in the legacy version. However, that approach was not viable anymore once we started testing the flow locally.

In the patch (#480) I am now taking the approach where we distinguish between test environment and all other environments.

**Comment by jgehrcke** [comment]
This has been fixed, and also the next problem of the same kind was fixed by #499. Closing!
